share/ruby-install/package_manager.sh

```diff
@@ -32,30 +33,29 @@ function set_package_manager()
 function install_packages()
 {
 	case "$package_manager" in
-		apt)	$sudo apt-get install -y "$@" || return $? ;;
-		dnf|yum)$sudo $package_manager install -y "$@" || return $?     ;;
-		port)   $sudo port install "$@" || return $?       ;;
-		pkg)	$sudo pkg install -y "$@" || return $?     ;;
-		brew)
-			local brew_owner="$(/usr/bin/stat -f %Su "$(command -v brew)")"
-			if [ "$(whoami)" -eq "$brew_owner" ]; then
-				brew install "$@" ||
-				brew upgrade "$@" || return $?
-			else
-				sudo -u "$brew_owner" brew install "$@" ||
-				sudo -u "$brew_owner" brew upgrade "$@" || return $?
-			fi
-			;;
 		pacman)
 			local missing_pkgs=($(pacman -T "$@"))
 
 			if (( ${#missing_pkgs[@]} > 0 )); then
-				$sudo pacman -S "${missing_pkgs[@]}" || return $?
+				installer -S "${missing_pkgs[@]}" || return $?
 			fi
 			;;
-		zypper) $sudo zypper -n in -l $* || return $? ;;
-		"")	warn "Could not determine Package Manager. Proceeding anyway." ;;
+		"")	    warn "Could not determine Package Manager. Proceeding anyway." ;;
+		brew)   installer install "$@" || installer upgrade "$@" || return $? ;;
```

Comment:
It doesn't make sense that we're checking `$package_manager` but have a generic `installer` function; which is kind of what `install_packages` should be. We should either define different `function install_packages()` based on the `if/elif command -v ...` checks:

    if command -v dnf >/dev/null; then
          function install_packages()
          {
              $sudo dnf install "$@"
          }

or auto-detect everything in `detect_package_manager` (ex: `$sudo`) which `install_packages` would then use. Also note that `$sudo` can be defined as `sudo -u "$brew_owner"`.

Reply:
`port -y`?